src/main/java/graphql/nadel/Nadel.java

```diff
@@ -97,47 +88,67 @@ public TypeDefinitionRegistry buildRegistry(ServiceDefinition serviceDefinition)
-//
-//
-//            String parentType = fieldTransformation.getParentDefinition().getName();
-//
-//            String originalFieldName = fieldDefinition.getName();
-//            Type targetType = assertNotNull(fieldTransformation.getTargetType());
-//            String targetTypeName = TypeInfo.typeInfo(targetType).getTypeName().getName();
-//            String newFieldName = fieldTransformation.getTargetName();
-//
-//            String queryField = targetTypeName.toLowerCase();
-//            SchemaNamespace targetNamespace = findNameSpace(targetTypeName);
-//            Link link = Link
-//                    .from(schemaNamespace, parentType, newFieldName, originalFieldName)
-//                    .to(targetNamespace, targetTypeName, queryField, "id")
-//                    .replaceFromField()
-//                    .build();
-//            result.get(schemaNamespace).add(link);
-//        });
-        return result;
+
+    private List<Link> createLinks(ServiceDefinition serviceDefinition) {
+        List<Link> links = new ArrayList<>();
+
+        SchemaNamespace schemaNamespace = assertNotNull(this.namespaceByService.get(serviceDefinition.getName()));
+
+        final List<FieldDefinitionWithParentType> defs = collectFieldTransformations(serviceDefinition);
+
+        for (FieldDefinitionWithParentType definition : defs) {
+            final FieldTransformation transformation = definition.field().getFieldTransformation();
+            if (transformation.getInnerServiceHydration() != null) {
+                final InnerServiceHydration hydration = transformation.getInnerServiceHydration();
+                final Link link = createHydrationLink(schemaNamespace, definition, hydration);
+                links.add(link);
+            } else if (transformation.getFieldMappingDefinition() != null) {
+                throw new InvalidDslException("Field mapping not implemented yet.", transformation.getSourceLocation());
+            }
+        }
+        return links;
+    }
+
+    private Link createHydrationLink(SchemaNamespace schemaNamespace, FieldDefinitionWithParentType definition,
+                                     InnerServiceHydration hydration) {
+        SchemaNamespace targetService = assertNotNull(this.namespaceByService.get(hydration.getServiceName()));
+        final FieldDefinition targetField = findTargetFieldForHydration(hydration);
+        //TODO: will not work for lists or non nullable types. does braid support this at all?
+        String targetTypeName = TypeInfo.typeInfo(targetField.getType()).getName();
+
+        //TODO: braid does not support multiple arguments, so we just take first
+        final String fromField = hydration.getArguments().entrySet().stream()
+                .map(e -> e.getValue().getInputName())
+                .findFirst()
+                .orElse(definition.field().getName());
+
+        return Link.from(schemaNamespace, definition.parentType(), definition.field().getName(), fromField)
```

Comment:
@andimarek fromField can only be a single field. Should we make changes in braid or maybe restrict DSL so that it only allows passing one field?

Reply:
lets leave it for now for one field and we will improve braid in the future